pkg/machines/components/stateIcon.jsx

```diff
@@ -0,0 +1,89 @@
+/*
+ * This file is part of Cockpit.
+ *
+ * Copyright (C) 2016 Red Hat, Inc.
+ *
+ * Cockpit is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 2.1 of the License, or
+ * (at your option) any later version.
+ *
+ * Cockpit is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+ */
+import React from 'react';
+import PropTypes from 'prop-types';
+import cockpit from 'cockpit';
+
+import { Button, Label, Popover } from '@patternfly/react-core';
+import { ErrorCircleOIcon, PendingIcon } from '@patternfly/react-icons';
+
+import {
+    rephraseUI,
+} from "../helpers.js";
+
+import "./stateIcon.scss";
+
+const _ = cockpit.gettext;
+
+export const StateIcon = ({ state, valueId, error, dismissError }) => {
+    if (state === undefined) {
+        return (<div />);
+    }
+    const stateMap = {
+        /* VM states */
+        running: { color: "green" },
+        crashed: { color: "red" },
+        dying: { color: "red" },
+        'Creating VM': { className: <PendingIcon /> },
+        'Creating VM installation': { className: <PendingIcon /> },
+
+        /* Storage Pool/Network States  */
+        active : { color: "green" },
+    };
+
+    const label = (
+        <>
+            {error &&
+            <Label color="red"
+                icon={<ErrorCircleOIcon />}
+                className="resource-state-text"
+                onClose={dismissError}
+                id={valueId}>
+                <>
+                    {_("Failed")}
+                    <Button variant="link" isInline>{_("view more...")}</Button>
+                </>
+            </Label>}
+            <Label color={stateMap[state] && stateMap[state].color}
+                   icon={stateMap[state] && stateMap[state].icon}
```

Comment:
`icon` is never defined. I guess because you defined icons as classNames?

Reply:
Right, it was meant to be icon above.
Therefore the icons were missing from there UI.
Currently it looks like:
![Screen Shot 2020-11-03 at 15 52 16](https://user-images.githubusercontent.com/14921356/98000482-d6e8cb00-1dec-11eb-91a4-527be22c1f2a.png)